lua/pl/permute.lua

```diff
@@ -78,4 +78,104 @@ function permute.table (a)
     return res
 end
 
+
+
+--- an iterator over all permutations of the elements of the given lists.
+-- @param ... list-like tables, they are nil-safe if a length-field `n` is provided
```

Comment:
I might be worth mentioning in this documentation where `n` comes from. Some of the tests set it manualy, but the docstring here might mention that `utils.pack()`  does this. Otherwise the demo of being `nil` safe is just a bit hard to follow.